backend/filejump/filejump.go

```diff
@@ -0,0 +1,1680 @@
+func (f *Fs) Put(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) (fs.Object, error) {
+	// fs.Logf(nil, "Put wurde aufgerufen")
+	remote := src.Remote()
+	size := src.Size()
+	modTime := src.ModTime(ctx)
+
+	// For unknown size uploads, we need to read the data first to determine the size
+	if size < 0 {
+		// Read all data into memory to determine size
+		data, err := io.ReadAll(in)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read data for unknown size upload: %w", err)
+		}
+		size = int64(len(data))
+		// Create a new reader from the data
+		in = strings.NewReader(string(data))
+	}
+
+	o, _, _, err := f.createObject(ctx, remote, modTime, size)
+	if err != nil {
+		return nil, err
+	}
+	return o, o.Update(ctx, in, src, options...)
+}
+
+// // Mkdir makes the directory (container, bucket)
+// //
+// // Shouldn't return an error if it already exists
+// func (f *Fs) Mkdir(ctx context.Context, dir string) error {
+// 	presignResult, err := CallJSON[PresignResult](f, ctx, "POST", "/folders", url.Values{
+// 		"name": []string{dir}
+// :
+// "test123"
+// parentId
+// :
+// 		"Filename":     []string{leaf},
+// 		"Mime":         []string{"application/octet-stream"},
+// 		"Disk":         []string{"uploads"},
+// 		"Size":         []string{strconv.FormatInt(size, 10)},
+// 		"Extension":    []string{"bin"},
+// 		"WorkspaceID":  []string{"0"},
+// 		"ParentID":     []string{directoryID},
+// 		"RelativePath": []string{""},
+// 	})
+
+// 	if err != nil {
+// 		// Fehlerbehandlung
+// 	}
+// }
+
+// Mkdir creates the container if it doesn't exist
+func (f *Fs) Mkdir(ctx context.Context, dir string) error {
+	// fs.Logf(nil, "Mkdir wurde aufgerufen")
+	_, err := f.dirCache.FindDir(ctx, dir, true)
+	return err
+}
+
+// purgeCheck removes the root directory, if check is set then it
+// refuses to do so if it has anything in
+func (f *Fs) purgeCheck(ctx context.Context, dir string, check bool) error {
+	// fs.Logf(nil, "purgeCheck wurde aufgerufen")
+	root := path.Join(f.root, dir)
+	if root == "" {
+		return errors.New("can't purge root directory")
+	}
+	dc := f.dirCache
+	rootID, err := dc.FindDir(ctx, dir, false)
+	if err != nil {
+		return err
+	}
+
+	// If check is true, verify the directory is empty before deleting
+	if check {
+		isEmpty := true
+		_, err := f.listAll(ctx, rootID, false, false, true, func(item *api.Item) bool {
+			isEmpty = false
+			return true // Stop on first item found
+		})
+		if err != nil {
+			return fmt.Errorf("failed to check if directory is empty: %w", err)
+		}
+		if !isEmpty {
+			return fs.ErrorDirectoryNotEmpty
+		}
+	}
+
+	type RequestDelete struct {
+		EntryIds      []int `json:"entryIds"`
+		DeleteForever bool  `json:"deleteForever"`
+	}
+
+	type ResultDelete struct {
+		Status string `json:"status,omitempty"`
+	}
+
+	iDir, _ := strconv.Atoi(rootID)
+	request := RequestDelete{
+		EntryIds:      []int{iDir},
+		DeleteForever: true,
```

Comment:
Generally backends default to storing deletes in the trash and use a `hard_delete` flag to override.

Eg

      --jottacloud-hard-delete   Delete files permanently rather than putting them into the trash
      --mega-hard-delete         Delete files permanently rather than putting them into the trash

We started doing it that way round after one user deleted all of their files by accident! So safety first, but overridable by the user.